subsys/net/l2/ieee802154/ieee802154_utils.h

```diff
@@ -278,112 +278,204 @@ static inline void ieee802154_radio_remove_pan_id(struct net_if *iface, uint16_t
+
+	default:
+		CODE_UNREACHABLE;
+	}
+}
 
 /**
- * MAC utilities
+ * @brief Calculates the PHY's turnaround time for the current channel page (see
+ * section 11.3, table 11-1, aTurnaroundTime) in PHY symbols.
+ *
+ * @details The PHY's turnaround time is used to calculate - among other
+ * parameters - the TX-to-RX turnaround time (see section 10.2.2) and the
+ * RX-to-TX turnaround time (see section 10.2.3).
+ *
+ * @param iface The interface for which the turnaround time should be calculated.
  *
- * Note: While MAC utilities may refer to PHY utilities,
- *       the inverse is not true.
+ * @returns The turnaround time for the given interface in symbols.
+ * @retval 0 if an error occurred.
  */
+static inline uint32_t ieee802154_radio_get_a_turnaround_time(struct net_if *iface)
+{
+	enum ieee802154_phy_channel_page channel_page =
+		ieee802154_radio_current_channel_page(iface);
 
-/**
- * The number of PHY symbols forming a superframe slot when the superframe order
- * is equal to zero, see sections 8.4.2, table 8-93, aBaseSlotDuration and 6.2.1.
- */
-#define IEEE802154_MAC_A_BASE_SLOT_DURATION 60U
+	if (!channel_page) {
+		return 0;
+	}
+
+	/* Section 11.3, table 11-1, aTurnaroundTime: "For the SUN [...] PHYs,
+	 * the value is 1 ms expressed in symbol periods, rounded up to the next
+	 * integer number of sym- bol periods using the ceiling() function. [...]
+	 * The value is 12 [symbol periods] for all other PHYs.
+	 */
+
+	if (channel_page == IEEE802154_ATTR_PHY_CHANNEL_PAGE_NINE_SUN_PREDEFINED) {
+		/* Current SUN FSK drivers only implement legacy IEEE 802.15.4g
+		 * 863 MHz (Europe) and 915 MHz (US ISM) bands, see IEEE
+		 * 802.15.4g, section 5.1, table 0. Once more bands are required
+		 * we need to request the currently active frequency band from
+		 * the driver.
+		 */
+		return IEEE802154_PHY_A_TURNAROUND_TIME_1MS(
+			IEEE802154_PHY_SUN_FSK_863MHZ_915MHZ_SYMBOL_PERIOD_NS);
+	}
+
+	return IEEE802154_PHY_A_TURNAROUND_TIME_DEFAULT;
+}
 
 /**
- * The number of slots contained in any superframe, see section 8.4.2,
- * table 8-93, aNumSuperframeSlots.
+ * @brief Verify if the given channel lies within the allowed range of available
+ * channels of the driver's currently selected channel page.
+ *
+ * @param iface The interface for which the turnaround time should be calculated.
+ * @param channel The channel to verify or IEEE802154_NO_CHANNEL
+ *
+ * @returns true if the channel is available, false otherwise
  */
-#define IEEE802154_MAC_A_NUM_SUPERFRAME_SLOTS 16U
+bool ieee802154_radio_verify_channel(struct net_if *iface, uint16_t channel);
 
 /**
- * The number of PHY symbols forming a superframe when the superframe order is
- * equal to zero, see section 8.4.2, table 8-93, aBaseSuperframeDuration.
+ * @brief Counts all available channels of the driver's currently selected
+ * channel page.
+ *
+ * @param iface The interface for which the turnaround time should be calculated.
```

Comment:
Please reword the documentation of this parameter here and in other functions (e.g. `ieee802154_radio_verify_channel`) where turnaround time is referred to out of context.

Reply:
Fixed.